src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -944,6 +944,155 @@ public virtual RedirectToRouteResult RedirectToRoutePermanent(string routeName,
+        /// </summary>
+        /// <param name="pageName">The name of the page.</param>
+        /// <param name="routeValues">The parameters for a route.</param>
+        /// <returns>The <see cref="RedirectToPageResult"/> with <see cref="RedirectToPageResult.Permanent"/> set.</returns>
+        [NonAction]
+        public virtual RedirectToPageResult RedirectToPagePermanent(string pageName, object routeValues)
+            => RedirectToPagePermanent(pageName, routeValues, fragment: null);
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status301MovedPermanently"/>) to the specified <paramref name="pageName"/>
+        /// using the specified <paramref name="fragment"/>.
+        /// </summary>
+        /// <param name="pageName">The name of the page.</param>
+        /// <param name="fragment">The fragment to add to the URL.</param>
+        /// <returns>The <see cref="RedirectToPageResult"/> with <see cref="RedirectToPageResult.Permanent"/> set.</returns>
+        [NonAction]
+        public virtual RedirectToPageResult RedirectToPagePermanent(string pageName, string fragment)
+            => RedirectToPagePermanent(pageName, routeValues: null, fragment: fragment);
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status301MovedPermanently"/>) to the specified <paramref name="pageName"/>
+        /// using the specified <paramref name="routeValues"/> and <paramref name="fragment"/>.
+        /// </summary>
+        /// <param name="pageName">The name of the page.</param>
+        /// <param name="routeValues">The parameters for a route.</param>
+        /// <param name="fragment">The fragment to add to the URL.</param>
+        /// <returns>The <see cref="RedirectToPageResult"/> with <see cref="RedirectToPageResult.Permanent"/> set.</returns>
+        [NonAction]
+        public virtual RedirectToPageResult RedirectToPagePermanent(string pageName, object routeValues, string fragment)
+            => new RedirectToPageResult(pageName, routeValues, permanent: true, fragment: fragment);
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status307TemporaryRedirect"/>) to the specified page with 
+        /// <see cref="RedirectToRouteResult.Permanent"/> set to false and <see cref="RedirectToRouteResult.PreserveMethod"/>
+        /// set to true, using the specified <paramref name="pageName"/>, <paramref name="routeValues"/>, and <paramref name="fragment"/>.
+        /// </summary>
+        /// <param name="pageName">The name of the page.</param>
+        /// <param name="routeValues">The route data to use for generating the URL.</param>
+        /// <param name="fragment">The fragment to add to the URL.</param>
+        /// <returns>The created <see cref="RedirectToRouteResult"/> for the response.</returns> 
+        [NonAction]
+        public virtual RedirectToPageResult RedirectToPagePreserveMethod(
+            string pageName = null,
```

Comment:
Make this required. It really doesn't make sense to have a `null` page from a Controller.